mdata/store_cassandra.go

```diff
@@ -290,6 +291,7 @@ func NewCassandraStore(addrs, keyspace, consistency, CaPath, Username, Password,
 		omitReadTimeout:  time.Duration(omitReadTimeout) * time.Second,
 		ttlTables:        ttlTables,
 		tracer:           opentracing.NoopTracer{},
+		timeout:          cluster.Timeout,
```

Comment:
note that we default to 1000 ms, this seems too aggressive. looking at some customer dashboards, some customers are seeing multi-second query durations (cassGetExecDuration)

Reply:
1) this timeout is only used for writes.  reads just use the existing requestContext, which we dont currently set a timeout on and instead just rely on the context being canceled, eg when the client disconnects.

2) queries will fail if they run longer then cluster.Timeout. 
 https://github.com/grafana/metrictank/blob/ca4a891be799a2aafb5f70ba10738eac9871050b/vendor/github.com/gocql/gocql/conn.go#L612-L658

So if we have queries that are running longer then this, then it will be because they are blocked waiting for a stream to become available.

Reply:
1. ah yes..
2. in insertChunk we call `context.WithTimeout(context.Background(), c.timeout)`
doesn't that mean that in the select either of the 2 operations (between lines 648 and 655 in the code shown above) can succeed at practically the same time?

one comes from the timeout set from `c.timeout` which is the same timeout value we specified in the config, whereas the ctxDone fires when the context's deadline triggers, which is a deadline we set also based on the same timeout?

IOW i don't see the use for the context on the insert queries since they seem to achieve the same as the already setup timeout mechanism. am i missing something?

Reply:
c.timeout is only measured from after the request is sent to cassandra.
the context.Deadline is set when the request is constructed, so includes the time the request sits waiting to be sent.

Reply:
let's rename this field to `writeTimeout` to avoid confusion

Reply:
> let's rename this field to writeTimeout to avoid confusion
No.  CassandraStore.timeout  is assigned the same value as the gocql cluster.Timeout, which is for reads and writes.

Variables should be named based on the information they contain, not on how they are used.